source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/FluentIconImageLoaderAsync.kt

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package io.adaptivecards.renderer
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.os.AsyncTask
+import android.view.View
+import android.widget.ImageView
+import com.caverock.androidsvg.SVG
+import io.adaptivecards.objectmodel.AdaptiveCardObjectModel
+import io.adaptivecards.renderer.http.HttpRequestHelper
+import io.adaptivecards.renderer.http.HttpRequestResult
+import org.json.JSONObject
+import java.lang.ref.WeakReference
+import java.nio.charset.StandardCharsets
+
+/**
+ * Responsible for rendering the Fluent Icon element
+ **/
+open class FluentIconImageLoaderAsync(
+    val renderedCard: RenderedAdaptiveCard,
+    val iconSize: Long,
+    private val iconColor: String,
+    view: View
+) : AsyncTask<String, Void, HttpRequestResult<String>>() {
```

Comment:
Can we use `Coroutine`? if it is a big change, will take care of it in an another PR.

Reply:
I used the existing infra and dependencies in the SDK. It'll be big refactoring. This can be taken up separately in another PR.